src/editor/map/map_context.cpp

```diff
@@ -539,7 +539,7 @@ config map_context::to_config()
 
 		side["fog"] = t->uses_fog();
 		side["shroud"] = t->uses_shroud();
-		side["share_vision"] = t->share_vision();
+		side["share_vision"] = team_shared_vision::get_string(t->share_vision());
```

Comment:
So just theoretically, what happens if you didn't make this change? Will it still compile but produce the wrong answer?

Reply:
It fails to compile because `enum class` isn't implicitly convertible to an int, and config doesn't have a separate assignment operator to use for this case.

Reply:
So basically it relies on the enum being scoped.

I guess that's fine as long as we note that the enum _must_ be scoped (I wonder if that's enforcible via `static_assert`?)

Reply:
https://stackoverflow.com/questions/15586163/c11-type-trait-to-differentiate-between-enum-class-and-regular-enum it looks like it would be something like: 
`static_assert(std::integral_constant<bool, std::is_enum<typename T::type>::value && !std::is_convertible<typename T::type, int>::value>::value, "Enum is not a scoped enum");`
which does correctly result in a compile error if an instance of it uses a regular enum.